pre_commit_hooks/no_commit_to_branch.py

```diff
@@ -10,14 +10,17 @@
 from pre_commit_hooks.util import cmd_output
 
 
-def is_on_branch(protected):  # type: (Set[str]) -> bool
+def is_on_branch(protected, patterns=set()):
```

Comment:
the default value here should be immutable (such as `frozenset()`)

Reply:
Cheers for the review :smile:! Have made the change. Also had to change the calling function to satisfy the type checker